07-final-project/views/character-info-view.js

```diff
@@ -0,0 +1,105 @@
+//VENDOR
+var React = require('react');
+var $ = require('jquery');
+//STORES
+var characterInfoStore = require('../stores/character-info-store');
+//ACTIONS
+var characterInfoStoreAction = require('../stores/character-info-store-actions');
+//COMPONENTS
+var Input = require('../components/input');
+var CharacterInfoList = require('../components/character-info-list');
+
+var CharacterInfoView = React.createClass({
+
+    getInitialState: function () {
+        return {
+            name: '',
+            realm: '',
+            serverData: characterInfoStore.getCharacterInfo()
+        }
+    },
+
+    componentWillMount: function () {
+        characterInfoStore.addActionListener(characterInfoStoreAction.CHANGE_EVENT, this.updateList);
+    },
+
+    componentWillUnmount: function () {
+        characterInfoStore.removeActionListener(characterInfoStoreAction.CHANGE_EVENT, this.updateList);
+    },
+
+    render: function () {
+
+        return (
+            <form {...this.getFormProps()}>
+                <Input {...this.getNameInputProps()} />
+                <Input {...this.getRealmInputProps()} />
+                <button>Search</button>
+                <CharacterInfoList {...this.getCharacterInfoListProps()}/>
+            </form>
+        );
+    },
+
+    getFormProps: function () {
+        return {
+            onSubmit: this.handleFormSubmit
+        }
+    },
+
+    getNameInputProps: function () {
+        return {
+            value: this.state.name,
+            onChange: this.handleNameInputChange
+        }
+    },
+
+    getRealmInputProps: function () {
+        return {
+            value: this.state.realm,
+            onChange: this.handleRealmImputChange
+        }
+    },
+
+    getCharacterInfoListProps: function () {
+        return {
+            data: this.state.serverData
+        }
+    },
+
+    handleNameInputChange: function (event) {
+        this.setState({name: event.target.value});
+    },
+
+    handleRealmImputChange: function (event) {
+        this.setState({realm: event.target.value});
+    },
+
+    handleFormSubmit: function (event) {
+        event.preventDefault();
+        this.getServerResponse(this.state.realm, this.state.name);
+        this.resetForm();
+    },
+
+    getServerResponse: function (realm, characterName) {
+        $.ajax({
```

Comment:
From my point of view you can create a new "something" to store all the request logic (with all the api calls)